codegen-v2/src/main.rs

```diff
@@ -4,17 +4,87 @@
 // terms governing use, modification, and redistribution, is contained in the
 // file LICENSE at the root of the source code distribution tree.
 
-use libparser::parse;
-use std::fs::File;
-use std::io::Write;
-use std::path::Path;
-
-fn main() {
-    let path = Path::new("../include/");
-    let dir = parse(&path).expect("Failed to parse path");
-    let data = serde_json::to_string_pretty(&dir.map).expect("Failed to generate JSON");
-
-    let mut file = File::create("include_manifest.json").expect("Unable to create file");
-    file.write_all(data.as_bytes())
-        .expect("Unable to write data to file");
+use libparser::codegen::swift::RenderIntput;
+use libparser::manifest::parse_dir;
+use libparser::{Error, Result};
+use std::fs::read_to_string;
+
+fn main() -> Result<()> {
+    let args: Vec<String> = std::env::args().collect();
+
+    if args.len() < 2 {
+        panic!("Invalid command");
+    }
+
+    match args[1].as_str() {
+        "swift" => generate_swift_bindings(),
+        _ => Err(Error::InvalidCommand),
+    }
+}
+
+fn generate_swift_bindings() -> Result<()> {
+    // NOTE: The paths will be configurable, eventually.
+    const OUT_DIR: &str = "bindings/";
+    std::fs::create_dir_all(OUT_DIR)?;
+
+    let struct_t = read_to_string("src/codegen/swift/templates/struct.hbs")?;
```

Comment:
What do you think about
```rust
let templates = std::path::PathBuf::from("src/codegen/swift/templates");

let enum_t = read_to_string(templates.join("enum.hbs"))?;
```
Or by using
```rust
const TEMPLATES: &str = "src/codegen/swift/templates";
let enum_t = read_to_string(&format!("{TEMPLATES}/enum.hbs"));
```
Whatever you prefer. It will be easier to change the path if needed

Reply:
Done!